packages/client/src/utils/index.ts

```diff
@@ -153,10 +155,22 @@ export async function buildFileTree(
       } catch (e) {
         debug(`Error detecting microfrontend config: ${e}`);
       }
+    }
 
-      if (refs.size > 0) {
-        fileList = fileList.concat(Array.from(refs));
+    // Check for .vercel/routes.json and include it if it exists
+    try {
+      const routesJsonPath = join(path, '.vercel', 'routes.json');
+      const routesJsonContent = await maybeRead(routesJsonPath, null);
```

Comment:
Do we actually care about the contents here? If not `fs.stat` is probably faster.